x-pack/plugins/observability/public/context/has_data_context.tsx

```diff
@@ -49,23 +55,53 @@ export function HasDataContextProvider({ children }: { children: React.ReactNode
       if (!isExploratoryView)
         apps.forEach(async (app) => {
           try {
-            if (app !== 'alert') {
-              const params =
-                app === 'ux'
-                  ? { absoluteTime: { start: absoluteStart, end: absoluteEnd } }
-                  : undefined;
-
-              const result = await getDataHandler(app)?.hasData(params);
-              setHasData((prevState) => ({
+            const updateState = ({
+              hasData,
+              indices,
+              serviceName,
+            }: {
+              hasData?: boolean;
+              serviceName?: string;
+              indices?: string | ApmIndicesConfig;
+            }) => {
+              setHasDataMap((prevState) => ({
                 ...prevState,
                 [app]: {
-                  hasData: result,
+                  hasData,
+                  ...(serviceName ? { serviceName } : {}),
+                  ...(indices ? { indices } : {}),
                   status: FETCH_STATUS.SUCCESS,
                 },
               }));
+            };
+            switch (app) {
```

Comment:
Seems like this could have a larger impact on the observability overview page. Should I smoke test the overview page as well?

Reply:
Yes, that would be helpful